src/Queue.h

```diff
@@ -21,90 +21,243 @@
 //	Entries must be either a pointer to the data or nonzero data with
 //	sizeof(data) <= sizeof(void*).
 
-#include "List.h"
+template<typename T>
+class QueueIterator
+	{
+	T* const entries;
+	int offset;
+	int num_entries;
+	T endptr; // let this get set to some random value on purpose. It's only used
+			  // for the operator[] and operator* cases where you pass something
+			  // off the end of the collection, which is undefined behavior anyways.
```

Comment:
I tried this when I was working on the original conversion. The operator has to return something and it can't be a temporary variable because it's returning a reference.

Reply:
I'm going to just remove the `endptr` and return an indexed value from the operators. There's the chance that we could return something off the end of the array, but that's also undefined behavior so it ends up the same.